compiler/modules/types/core.bal

```diff
@@ -299,7 +297,11 @@ function unaryTypeCheckOpPanic(Context cx, SubtypeData t) returns boolean {
 type BasicTypeOps readonly & record {|
     BinOp union = binOpPanic;
     BinOp intersect = binOpPanic;
-    BinOp diff = binOpPanic;
+    // diff can be implemented combining intersection and complement. i.e. diff = intersect(d1, complement(d2))
+    // () is used when we want the aformentioned formular as the diff. 
+    BinOp? diff = binOpPanic;
```

Comment:
can we make this 
```suggestion
    BinOp? diff = ();
```
Then you can get rid of `diff: (),` from subtypes.

Reply:
+1

Reply:
Can get rid of `binOpPanic` function too.

Reply:
`binOpPanic` is used as the defaultable function for union and intersection. Since they do not have a default implementation it is okay to have `binOpPanic` as default right?

Reply:
Ya, I made a mistake I though we have different panic functions for each. Can't get rid of binOpPanic